library/std/src/net/ip.rs

```diff
@@ -1488,6 +1488,36 @@ impl Ipv6Addr {
         (self.segments()[0] & 0xff00) == 0xff00
     }
 
+    /// Converts this address to an [IPv4 address] if it's an "IPv4-mapped IPv6 address"
+    /// defined in [IETF RFC 4291 section 2.5.5.2], otherwise returns [`None`].
+    ///
+    /// ::ffff:a.b.c.d becomes a.b.c.d
```

Comment:
This paragraph should still be improved. See my suggestion [here](https://github.com/rust-lang/rust/pull/75019#discussion_r467464194)